google-cloud-util/google-cloud-compat-checker/src/main/java/com/google/cloud/compatchecker/GoogleCloudCompatChecker.java

```diff
@@ -43,30 +44,96 @@ protected void logProperty(String s, String s1) {
   }
 
   public static void main(String[] args) {
+    Boolean checkResult = check();
+    if (checkResult != null && !checkResult) {
+      System.exit(1);
+    }
+  }
+
+  public static Boolean check() {
```

Comment:
I don't think there is value in splitting this yet... it's just a straight run of checks; putting structure on this requires more design thought than I think is merited for this problem.